fdbserver/ptxn/test/RealStorageServer.actor.cpp

```diff
@@ -0,0 +1,180 @@
+		std::cout << "new Storage Server file name: " << fileName << std::endl;
+		deleteFile(fileName);
+		state IKeyValueStore* data = openKVStore(storeType, fileName, ssi.id(), 0);
+
+		state ReplyPromise<InitializeStorageReply> storageReady;
+
+		// Initialize dbInfo with mocked cursor and mocked loy system.
+
+		// TODO: Use Reference API?
+		// TODO: delete mockLogSystem eventually.
+		state MockLogSystem* mockLogSystem = new MockLogSystem();
+		mockLogSystem->cursor = self->mockPeekCursor.castTo<ILogSystem::IPeekCursor>();
+
+		LogSystemConfig logSystemConfig;
+		logSystemConfig.logSystemType = LogSystemType::mock;
+		logSystemConfig.mockLogSystem = mockLogSystem;
+
+		ServerDBInfo dbInfoBuilder;
+		dbInfoBuilder.recoveryState = RecoveryState::ACCEPTING_COMMITS;
+		dbInfoBuilder.logSystemConfig = logSystemConfig;
+		state Reference<AsyncVar<ServerDBInfo>> dbInfo = makeReference<AsyncVar<ServerDBInfo>>(dbInfoBuilder);
+		// Initialize dbInfo done.
+
+		std::cout << "Starting Storage Server." << std::endl;
+		state Future<Void> ss = storageServer(data, ssi, self->tag, storageReady, dbInfo, folder);
+		std::cout << "Storage Server started." << std::endl;
+
+		self->actors.add(ss);
+		wait(ss);
+
+		return Void();
+	}
+
+	ACTOR static Future<Void> verifyGetValue(StorageServerTestDriver* self,
+	                                         KeyRef key,
+	                                         Version version,
+	                                         ValueRef expectedValue) {
+		wait(switchToServerProcess(self));
+
+		GetValueRequest getValueRequest = GetValueRequest(UID(), // spanContext
+		                                                  key,
+		                                                  version,
+		                                                  Optional<TagSet>(), // tags
+		                                                  Optional<UID>()); // debugID
+		self->ssi.getValue.send(getValueRequest);
+
+		GetValueReply getValueReply = wait(getValueRequest.reply.getFuture());
+		const Value& value = getValueReply.value.get();
+		std::cout << "Get value: " << value.toString() << std::endl;
+		ASSERT(value == expectedValue);
+
+		wait(switchBack(self));
+		return Void();
+	}
+};
+} // namespace ptxn
+
+TEST_CASE("fdbserver/ptxn/test/storageserver") {
+	using namespace ptxn;
+
+	state ptxn::StorageServerTestDriver driver;
+
+	// Set the mock cursor with some input
+	Arena arena;
+	MutationRef mutation(arena, MutationRef::SetValue, "Hello"_sr, "World"_sr);
+	std::string message = BinaryWriter::toValue(mutation, AssumeVersion(currentProtocolVersion)).toString();
+	std::vector<Tag> tags = { driver.tag };
+	state std::vector<MockPeekCursor::OneVersionMessages> allVersionMessages = { MockPeekCursor::OneVersionMessages(
+		8, // version
+		{ MockPeekCursor::MessageAndTags(message, tags) }) };
+	driver.mockPeekCursor = makeReference<MockPeekCursor>(allVersionMessages, arena);
```

Comment:
As an example of how taking references can be subtle, the lifetime of the `mockPeekCursor` here is longer than the lifetime of `allVersionMessages`. I don't think there is an actual bug, but were `MockPeekCursor` ever to touch `allVersionMessages` in its destructor, there would be a use after free bug.